smtpclient/SMTPClient.cpp

```diff
@@ -269,107 +269,109 @@ const std::string SMTPClient::MakeMessage()
-		ret += '\n';
+		ret += "--" + std::string(szBoundaryMixed) + "\r\n";
+		ret += "Content-type: text/plain; charset=utf-8; format=flowed\r\n"
+			"Content-transfer-encoding: 7bit\r\n\r\n";
+		ret += "\r\n";
 	}
 
 	if (!m_Attachments.empty())
 	{
-		ret += '\n';
+		ret += "\r\n";
 		// now add each attachment.
 		std::vector<std::pair<std::string, std::string> >::const_iterator it1;
 		for (it1 = m_Attachments.begin(); it1 != m_Attachments.end(); ++it1)
 		{
-			ret += "--" + std::string(szBoundaryMixed) + '\n';
+			ret += "--" + std::string(szBoundaryMixed) + "\r\n";
 			if (it1->second.length() > 3)
 			{ // long enough for an extension
 				std::string typ(it1->second.substr(it1->second.length() - 4, 4));
 				if (typ == ".gif") { // gif format presumably
-					ret += "Content-Type: image/gif;\n";
+					ret += "Content-Type: image/gif;\r\n";
 				}
 				else if (typ == ".jpg" || typ == "jpeg") { // j-peg format presumably
-					ret += "Content-Type: image/jpg;\n";
+					ret += "Content-Type: image/jpg;\r\n";
 				}
 				else if (typ == ".txt") { // text format presumably
-					ret += "Content-Type: plain/txt;\n";
+					ret += "Content-Type: plain/txt;\r\n";
 				}
 				else if (typ == ".bmp") { // windows bitmap format presumably
-					ret += "Content-Type: image/bmp;\n";
+					ret += "Content-Type: image/bmp;\r\n";
 				}
 				else if (typ == ".htm" || typ == "html") { // hypertext format presumably
-					ret += "Content-Type: plain/htm;\n";
+					ret += "Content-Type: plain/htm;\r\n";
 				}
 				else if (typ == ".png") { // portable network graphic format presumably
-					ret += "Content-Type: image/png;\n";
+					ret += "Content-Type: image/png;\r\n";
 				}
 				else if (typ == ".exe") { // application
-					ret += "Content-Type: application/X-exectype-1;\n";
+					ret += "Content-Type: application/X-exectype-1;\r\n";
 				}
 				else { // add other types
 					   // everything else
-					ret += "Content-Type: application/X-other-1;\n";
+					ret += "Content-Type: application/X-other-1;\r\n";
 				}
 			}
 			else {
 				// default to don't know
-				ret += "Content-Type: application/X-other-1;\n";
+				ret += "Content-Type: application/X-other-1;\r\n";
 			}
 
-			ret += "\tname=\"" + it1->second + "\"\n";
-			ret += "Content-Transfer-Encoding: base64\n";
-			ret += "Content-Disposition: attachment;\n filename=\"" + it1->second + "\"\n\n";
+			ret += "\tname=\"" + it1->second + "\"\r\n";
+			ret += "Content-Transfer-Encoding: base64\r\n";
+			ret += "Content-Disposition: attachment;\r\n filename=\"" + it1->second + "\"\r\n\r\n";
 
 			ret.insert(ret.end(), it1->first.begin(), it1->first.end());
-			ret += '\n';
+			ret += "\r\n";
 		}
-		ret += "--" + std::string(szBoundaryMixed) + "--\n";
+		ret += "--" + std::string(szBoundaryMixed) + "--\r\n";
 	}
 
-	// end the data in the message.
-	ret += "\n.\n";
+	// We do not have to end the message with "<CRLF>.<CRLF>" as libcurl does this for us
+	// ret += "\r\n.\r\n";
+	
+	ret += "\r\n";
```

Comment:
I am 99,999% sure this last \r\n is no needed, can you verify this

Reply:
libcurl just sends <dot>CRLF so my code is defensive in case anything sends a plain text mail which doesn't terminate with a CRLF.  If everything is sent MIME encoded, and the MIME encoding is performed by SMTPClient.cpp then you are correct that the `ret += "\r\n";` is not needed, and I am very happy for you to remove it.

Reply:
If possible please test it